examples/allocation/zig/testdata/greet.zig

```diff
@@ -0,0 +1,79 @@
+const std = @import("std");
+const allocator = std.heap.page_allocator;
+
+// main is required for Zig to compile to enable WASI.
+pub fn main() anyerror!void {
+    return;
+}
+
+extern "env" fn log(ptr: u32, size: u32) void;
+
+// _log prints a message to the console using log.
+pub fn _log(message: []u8) void {
+    const ptrSize = stringToPtr(message);
+    var p = ptrSize >> 32;
+    log(@truncate(u32, p), @truncate(u32, ptrSize));
+}
+
+pub export fn malloc(size: usize) usize {
+    var memory = allocator.alloc(u8, size) catch unreachable;
+    return @ptrToInt(memory.ptr);
+}
+
+pub export fn free(ptr: usize) void {
+    var s: []u8 = undefined;
+    s.ptr = @intToPtr([*]u8, ptr);
+    allocator.free(s);
+}
+
+pub fn _greeting(name: []u8) ![]u8 {
+    return try std.fmt.allocPrint(
+        allocator,
+        "Hello, {s}!",
+        .{name},
+    );
+}
+
+// _greet prints a greeting to the console.
+pub fn _greet(name: []u8) void {
```

Comment:
It would be more idiomatic to make this accept `[]const u8` as you're not mutating the slice.

Reply:
Done
https://github.com/tetratelabs/wazero/pull/741/commits/828c95303d4a37e3b0d0d6ffa8968e2630965787